src/core/mac/sub_mac.cpp

```diff
@@ -276,6 +276,52 @@ void SubMac::CslSample(void)
 }
 #endif // OPENTHREAD_CONFIG_MAC_CSL_RECEIVER_ENABLE
 
+#if OPENTHREAD_CONFIG_MAC_CSL_DEBUG_ENABLE
+void SubMac::LogReceived(RxFrame *aFrame)
+{
+    Address  dst;
+    char     logString[64];
+    int32_t  drift;
+    uint32_t sampleTime, ahead, after;
+
+    IgnoreError(aFrame->GetDstAddr(dst));
+
+    VerifyOrExit((dst.GetType() == Address::kTypeShort && dst.GetShort() == GetShortAddress()) ||
+                 (dst.GetType() == Address::kTypeExtended && dst.GetExtended() == GetExtAddress()));
+
+    LogWarn("Received frame in state (SubMac %s, CSL %s), timestamp %lu", StateToString(mState),
+            mIsCslSampling ? "CslSample" : "CslSleep",
+            ToUlong(static_cast<uint32_t>(aFrame->mInfo.mRxInfo.mTimestamp)));
+
+    VerifyOrExit(mState == kStateCslSample);
+
+    GetCslWindowEdges(ahead, after);
+    ahead -= kCslReceiveTimeAhead;
+
+    sampleTime = RadioSupportsReceiveTiming() ? mCslSampleTime.GetValue() - mCslPeriod * kUsPerTenSymbols
+                                              : mCslSampleTime.GetValue();
+
+    drift = aFrame->mInfo.mRxInfo.mTimestamp + kRadioHeaderPhrDuration - sampleTime;
+    snprintf(logString, sizeof(logString), "Target sample time %lu, margin %lu, time drift %d", ToUlong(sampleTime),
+             ToUlong(ahead), drift);
+
+    // Treat as a warning when the drift is not within the margins. Neither kCslReceiveTimeAhead
+    // or kMinCslWindow are considered for the margin since they have no impact on understanding
+    // possible drift errors between transmitter and receiver. So in this case ahead equals after.
+    if ((drift + ahead > 0) && (drift < static_cast<int32_t>(ahead)))
+    {
+        LogWarn("%s", logString);
+    }
+    else
+    {
+        LogCrit("%s", logString);
```

Comment:
would expect LogWarn() here? "Crit" means something critical to the integrity of the device itself; which isn't the case here - it's the other party that presumably made a mistake.